numpy/core/tests/test_multiarray.py

```diff
@@ -3548,21 +3566,55 @@ def test_test_zero_rank(self):
 
 
 class TestPickling(object):
+    def test_highest_available_pickle_protocol(self):
+        try:
+            import pickle5
+        except ImportError:
+            pickle5 = None
+
+        if sys.version_info[:2] >= (3, 8) or pickle5 is not None:
+            assert pickle.HIGHEST_PROTOCOL >= 5
+        else:
+            assert pickle.HIGHEST_PROTOCOL < 5
+
+    def test_array_with_dtype_has_object_pickling(self):
+        if pickle.HIGHEST_PROTOCOL < 5:
+            pass
+        else:
```

Comment:
Instead of passing the test, this should be skipped with pytest:

```python
@pytest.mark.skipif(pickle.HIGHEST_PROTOCOL < 5, reason="requires pickle protocol 5")
def test_array_with_dtype_has_object_pickling(self):
     my_object = object()
     ...